java-client/src/main/java/org/opensearch/client/transport/endpoints/SimpleEndpoint.java

```diff
@@ -60,21 +60,41 @@ public static <T> Function<T, Map<String, String>> emptyMap() {
     private final Function<RequestT, Map<String, String>> headers;
     private final boolean hasRequestBody;
     private final JsonpDeserializer<ResponseT> responseParser;
+    private final JsonpDeserializer<ErrorResponseT> errorDeserializer;
+
+    public SimpleEndpoint(
```

Comment:
Let's write one constructor with fewer parameters to reuse the other with more parameters?

Reply:
Done.